### ISSUE TITLE
Bug about HE_SIGNATURE_DECLARES_HASHING_OF_UNHASHABLE_CLASS 

### ISSUE THREAD

**Issue by LynnBroe** [spotbugs/spotbugs#2402]
TestA.java:
```java
import java.util.HashMap;
public class TestA {
    static class UMap extends HashMap<TestA, String> {}
    ;
    static HashMap<TestA, String> m =
            new HashMap<TestA, String>();

    static int foo(HashMap<TestA, String> map) {
        return map.size();
    }
    @Override
    public boolean equals(Object o) {
        return hashCode() == o.hashCode();
    }
    public static String add(TestA b, String s) {
        return m.put(b, s);
    }
    public static String get(TestA b) {
        return m.get(b);
    }
}
```
SpotBugs reported a `HE_SIGNATURE_DECLARES_HASHING_OF_UNHASHABLE_CLASS` warning in the `testA.java` file above. But if I use SpotBugs to detect some same`TestA.java` files, the results are not equivalent. SpotBugs can only report this warning in part of files. I think it is a bug. Analysis reports of each same code file should be equivalent.

Please refer to the MCVE [project](https://github.com/LynnBroe/spotbugs_test_HE_1).

**Comment by LynnBroe** [comment]
`HE_USE_OF_UNHASHABLE_CLASS` has a similar problem
